{
    "active_lock_reason": null,
    "assignee": null,
    "assignees": [],
    "author_association": "NONE",
    "body": "### Problem\n\n➜  remote-ssh.nvim git:(main) ✗ nvim --clean --server 139.59.38.127:6666 --remote-ui\nRemote ui failed to start: connection refused\n[1]    568404 segmentation fault (core dumped)  nvim --server 139.59.38.127:6666 --remote-ui\n\n\n### Steps to reproduce\n\non server:\n```bash\nnvim --clean --listen 0.0.0.0:6666\n```\n\non local:\n```bash\n➜  remote-ssh.nvim git:(main) ✗ nvim --clean --server 139.59.38.127:6666 --remote-ui\nRemote ui failed to start: connection refused\n[1]    568962 segmentation fault (core dumped)  nvim --clean --server 139.59.38.127:6666 --remote-ui\n\n```\n\nI tried with local using port forwarding working fine only faced issue with [ server and local ].\n\nthis is working:\n```bash\n# first terminal \nnvim --listen 0.0.0.0:6666\n\n# second terminal\nnvim --server 0.0.0.0:6666 --remote-ui\n\n```\n\n### Expected behavior\n\nit should open a remote ui attached to `--listen` nvim session\n\n### Nvim version (nvim -v)\n\nserver [NVIM v0.11.0-dev-1953+gafc3c43f12] local [NVIM v0.11.0-dev-1954+gc9b5ac113a-dirty]\n\n### Vim (not Nvim) behaves the same?\n\nnot checked\n\n### Operating system/version\n\non server [ ubuntu 24.4] on local [arch linux]\n\n### Terminal name/version\n\nkitty\n\n### $TERM environment variable\n\nxterm-kitty\n\n### Installation\n\nbuild from repo",
    "closed_at": null,
    "closed_by": null,
    "comment_data": [
        {
            "author_association": "NONE",
            "body": "I tried debugging it, but couldn't conclude anything ( first timer )\n\nAdding my observations.\n\n* `bt` after crash gives\n```\n* thread #1, name = 'nvim', stop reason = signal SIGSEGV: invalid permissions for mapped object (fault address: 0x7fffffffc750)\n  * frame #0: 0x00007fffffffc750\n    frame #1: 0x0000555555a736d4 nvim`uv__stream_destroy(stream=0x0000555555c03ed8) at stream.c:461:5\n    frame #2: 0x0000555555a628cc nvim`uv__finish_close(handle=0x0000555555c03ed8) at core.c:334:7\n    frame #3: 0x0000555555a629ce nvim`uv__run_closing_handles(loop=0x0000555555bd45e0) at core.c:364:5\n    frame #4: 0x0000555555a62cca nvim`uv_run(loop=0x0000555555bd45e0, mode=UV_RUN_DEFAULT) at core.c:462:5\n    frame #5: 0x00005555557159a6 nvim`loop_close(loop=0x0000555555bd45e0, wait=true) at loop.c:166:5\n    frame #6: 0x00005555557a87cf nvim`event_teardown at main.c:184:10\n    frame #7: 0x00005555557a9748 nvim`os_exit(r=1) at main.c:675:8\n    frame #8: 0x00005555557a9fb9 nvim`remote_request(params=0x00007fffffffca80, remote_args=4, server_addr=\"8.8.8.8:1234\", argc=5, argv=0x00007fffffffccf8, ui_only=true) at main.c:927:7\n    frame #9: 0x00005555557a8cd2 nvim`main(argc=5, argv=0x00007fffffffccf8) at main.c:343:5\n    frame #10: 0x00007ffff7c2a1ca libc.so.6`__libc_start_call_main(main=(nvim`main at main.c:250:1), argc=5, argv=0x00007fffffffccf8) at libc_start_call_main.h:58:16\n    frame #11: 0x00007ffff7c2a28b libc.so.6`__libc_start_main_impl(main=(nvim`main at main.c:250:1), argc=5, argv=0x00007fffffffccf8, init=<unavailable>, fini=<unavailable>, rtld_fini=<unavailable>, stack_end=0x00007fffffffcce8) at libc-start.c:360:3\n    frame #12: 0x00005555555ec705 nvim`_start + 37\n```\n\n* `0x00007fffffffc750` is address of `stream->connect_req->cb` in [libuv/src/unix/stream.c#uv__stream_destroy](https://github.com/weolar/miniblink49/blob/2443d477941b6189906882772307e15e5471ba43/node/uv/src/unix/stream.c#L443)\n\n* Looking at `cat /proc/<pid>/maps`, I find following\n   ```\n   7ffffffdd000-7ffffffff000 rw-p 00000000 00:00 0                          [stack]\n   ```\n   suggesting that this address lives on stack.\n\n\nHowever, I couldn't find where it is being set or what is the cause of this.\n\nIf someone finds this information useful, let me know how you proceed from here or how you find the bug/solution.",
            "created_at": "2025-03-15T13:10:49Z",
            "html_url": "https://github.com/neovim/neovim/issues/32883#issuecomment-2726540879",
            "id": 2726540879,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/32883",
            "node_id": "IC_kwDOAPphoM6ig7ZP",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/2726540879/reactions"
            },
            "updated_at": "2025-03-15T13:10:49Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/2726540879",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/19588421?v=4",
                "events_url": "https://api.github.com/users/80avin/events{/privacy}",
                "followers_url": "https://api.github.com/users/80avin/followers",
                "following_url": "https://api.github.com/users/80avin/following{/other_user}",
                "gists_url": "https://api.github.com/users/80avin/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/80avin",
                "id": 19588421,
                "login": "80avin",
                "node_id": "MDQ6VXNlcjE5NTg4NDIx",
                "organizations_url": "https://api.github.com/users/80avin/orgs",
                "received_events_url": "https://api.github.com/users/80avin/received_events",
                "repos_url": "https://api.github.com/users/80avin/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/80avin/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/80avin/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/80avin",
                "user_view_type": "public"
            }
        }
    ],
    "comments": 1,
    "comments_url": "https://api.github.com/repos/neovim/neovim/issues/32883/comments",
    "created_at": "2025-03-14T04:08:17Z",
    "events_url": "https://api.github.com/repos/neovim/neovim/issues/32883/events",
    "html_url": "https://github.com/neovim/neovim/issues/32883",
    "id": 2919020244,
    "labels": [
        {
            "color": "F9D0C4",
            "default": false,
            "description": "issue reporting a crash or segfault",
            "id": 435854234,
            "name": "bug-crash",
            "node_id": "MDU6TGFiZWw0MzU4NTQyMzQ=",
            "url": "https://api.github.com/repos/neovim/neovim/labels/bug-crash"
        },
        {
            "color": "c5def5",
            "default": false,
            "description": "remote UI, --remote commands, p2p / peer-to-peer",
            "id": 2269137131,
            "name": "remote",
            "node_id": "MDU6TGFiZWwyMjY5MTM3MTMx",
            "url": "https://api.github.com/repos/neovim/neovim/labels/remote"
        }
    ],
    "labels_url": "https://api.github.com/repos/neovim/neovim/issues/32883/labels{/name}",
    "locked": false,
    "milestone": null,
    "node_id": "I_kwDOAPphoM6t_LbU",
    "number": 32883,
    "performed_via_github_app": null,
    "reactions": {
        "+1": 0,
        "-1": 0,
        "confused": 0,
        "eyes": 1,
        "heart": 0,
        "hooray": 0,
        "laugh": 0,
        "rocket": 0,
        "total_count": 1,
        "url": "https://api.github.com/repos/neovim/neovim/issues/32883/reactions"
    },
    "repository_url": "https://api.github.com/repos/neovim/neovim",
    "state": "open",
    "state_reason": null,
    "sub_issues_summary": {
        "completed": 0,
        "percent_completed": 0,
        "total": 0
    },
    "timeline_url": "https://api.github.com/repos/neovim/neovim/issues/32883/timeline",
    "title": "nvim segmentation fault when connecting Remote UI via --server",
    "updated_at": "2025-03-15T13:10:49Z",
    "url": "https://api.github.com/repos/neovim/neovim/issues/32883",
    "user": {
        "avatar_url": "https://avatars.githubusercontent.com/u/67893205?v=4",
        "events_url": "https://api.github.com/users/kumarvivek1752/events{/privacy}",
        "followers_url": "https://api.github.com/users/kumarvivek1752/followers",
        "following_url": "https://api.github.com/users/kumarvivek1752/following{/other_user}",
        "gists_url": "https://api.github.com/users/kumarvivek1752/gists{/gist_id}",
        "gravatar_id": "",
        "html_url": "https://github.com/kumarvivek1752",
        "id": 67893205,
        "login": "kumarvivek1752",
        "node_id": "MDQ6VXNlcjY3ODkzMjA1",
        "organizations_url": "https://api.github.com/users/kumarvivek1752/orgs",
        "received_events_url": "https://api.github.com/users/kumarvivek1752/received_events",
        "repos_url": "https://api.github.com/users/kumarvivek1752/repos",
        "site_admin": false,
        "starred_url": "https://api.github.com/users/kumarvivek1752/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/kumarvivek1752/subscriptions",
        "type": "User",
        "url": "https://api.github.com/users/kumarvivek1752",
        "user_view_type": "public"
    }
}